### ISSUE TITLE
AntHocNet Link Failures Only

### ISSUE THREAD

**Issue by blackknight86** [Sawchord/anthocnet#1]
Greetings,

I tried to run "anthocnet-sim" example and noticed that I've got all experiments results unset (or 0 for PDR and Delay). Packets were sent but weren't received by any node. And type of every packet was set to link failure. I'm using ns3-dev version and Uubntu 16.04 host.
Maybe you can help me with my problem? I'll be very grateful!

Thanx a lot,
blackknight86

**Comment by bseker** [comment]
Greetings,

I am experiencing the same problem. Unfortunately, I only sees Link Failures. I tried to fix this but I couldnt. If it is fixed, pls let me know.

I am using Ubuntu 18.04 and NS30.1.

Thanks a lot
Burak

**Comment by Sawchord** [comment]
Hello, I suspect this might be due to a change in ns-3.27, where broadcast messages are not delivered to unicast addresses anymore out of the box.

Have you tried with ns-3.26?

**Comment by bseker** [comment]
Nope, didn't tried with ns-3.26. I will give a shot and post the result. 

Thanks for advice and thanks for your time.
Sincerely
Burak

**Comment by bseker** [comment]
In the changelog of ns-3.27(https://www.nsnam.org/releases/ns-3-27/), the following is stated:

> Sockets do not receive anymore broadcast packets, unless they are bound to an "Any" address (0.0.0.0) or to a subnet-directed broadcast packet (e.g., x.y.z.0 for a /24 noterok). As in Linux, the following rules are now enforced:
> A socket bound to 0.0.0.0 will receive everything.
> A socket bound to x.y.z.0/24 will receive subnet-directed broadcast (x.y.z.255) and unicast packets.
> A socket bound to x.y.z.w will only receive unicast packets.

I suppose, this is the update you were talking about. Leave this info here just to make the conversation more compact.

**Comment by bseker** [comment]
Hi again,

As sawchord mentined, the 'anthocnet-sim' looks ok with NS-3_26. The following are the results I get:

> PDR Delay Delay_Jitter Packet_overhead Byte_Overhead
> 0.947262
> 9.71382
> 228.948
> 3.59716
> 2.54669

Thanks for your help Sawchord. 

If I had time, maybe I can fix the issue for NS-3_27 and later releases. Time will tell, I will contact you if you want.

Have a good day,
Sincerely
Burak

**Comment by zhangyu683** [comment]
Yee，i use ns3.25, get the result. PDR is not 0.
